roles/jd-client/src/lib/job_declarator/mod.rs

```diff
@@ -55,7 +55,7 @@ pub struct JobDeclarator {
     req_ids: Id,
     min_extranonce_size: u16,
     // (Sent DeclareMiningJob, is future, template id, merkle path)
-    last_declare_mining_job_sent: Option<LastDeclareJob>,
+    last_declare_mining_jobs_sent: HashMap<u32, Option<LastDeclareJob>>,
```

Comment:
now that we are introducing a `HashMap`, I guess it makes sense to get rid of the `Option`?

I mean, there will never be a situation where we have a `(u32, None)` entry on this `HashMap`, right?
there's either a `(key, value)` entry, or there is not.

Reply:
make sure to not leak memory through this hash map.

Reply:
> now that we are introducing a `HashMap`, I guess it makes sense to get rid of the `Option`?
> 
> I mean, there will never be a situation where we have a `(u32, None)` entry on this `HashMap`, right? there's either a `(key, value)` entry, or there is nothing.

Yeah, but I preferred to not change too many things, because I think that the Option was put from someone else in the past to better manage some cases that I did not explore in this PR.

Reply:
this is not enough you have to remove things from it. When you initialize with capacity it just means that it allocate memory to get 10 elements. When you add the 11 you just reallocate.

Reply:
> this is not enough you have to remove things from it. When you initialize with capacity it just means that it allocate memory to get 10 elements. When you add the 11 you just reallocate.

Ok, once stabilized the fix I'm working on, I will push this changes, adding checks and reallocation of this hashmap

Reply:
@Fi3 done in 47dc7aa514740f6ac7c2c8065834a3082c75d714